### PR TITLE
Added a guideline to the source code

### PR DESCRIPTION
**Pull request by Yoric**
This should help with our effort towards packaging for Node.

### CODE REVIEW

File: LAYOUT.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- `crates/binjs_shared` contains shared utilities, used both at build-time and run-time. Features:
+  - tools for dealing with JSON;
+  - tools for dealing with ASTs
+- `crates/binjs_meta` is used mostly at build-time, to generate eg. the ES6 grammar or,
+  in Firefox, to generate the SpiderMonkey parser. It may also be used at runtime.
+  Features:
+  - read a grammar specification from a webidl;
+  - reflection of the grammar;
+  - sanity tests on the grammar;
+  - deanonymization of anonyous grammar nodes;
```
Comment (arai-a) [inline LAYOUT.md]: anonymous

File: LAYOUT.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  tools are slower and less safe than those of `crates/binjs_es6` but offer
+  a second implementation for testing purposes. Features:
+  - generating random JSON ASTs for a grammar;
+  - comparing two JSON ASTs in a grammar;
+  - encoding/decoding entire JSON ASTs from/to bytes using a format defined in `crates/binjs_io`.
+- `crates/binjs_es6` offers specialized tools to manipulate ES6 ASTs. Some of these
+  tools are generated by `crates/binjs_generate_library`. Features:
+  - a strongly-typed AST guaranteed to match the specs of `spec/es6.webidl`;
+  - annotating scopes, `eval`, ...
+  - serialization and deserialization of strongly-typed ES6 ASTs from/to bytess using a format defined in `crates/binjs_io`.
```
Comment (arai-a) [inline LAYOUT.md]: bytes

**Comment by Yoric** [comment]
Thanks for the quick review :)

**Comment by sendilkumarn** [comment]
Thanks, @Yoric for writing this down. 👍

**Pull request merged**